Unit Tests and Namespaces

NightAngel47 opened the pull request:
Created some play mode unit tests for CharacterMovement and MoonlightBehaviour. Also, created a root namespace and moved scripts into namespaces as well.

NightAngel47 committed:
Started work on adding Unit Tests

Started work on adding tests for existing code base. Current test is testing character movement in play mode as it is a monobehaviour script.

NightAngel47 committed:
Removed EditMode Tests

There was a build error since there were no edit mode tests yet.

NightAngel47 committed:
Moved Editor Folder out of Scripts Folder

Can't have an Editor folder in the scripts folder because of the Scripts Assemply Definition that is required for testing. Thus, the Editor folder has been moved under the Assets folder.

NightAngel47 committed:
Organized Scripts with Namespaces

Trying to do some organization.

NightAngel47 committed:
Moonlight Unity Tests

Added tests for moonlight being active and using the timer.

clinealex10 reviewed (changes requested):
I can walk through walls, not sure what that's from.
Also, they're a Debug.Log in MoonlightBehaviour.cs/ln46.

NightAngel47 committed:
Fixed Collision Issues by Adding Missing Layers

There were missing layers that were used to detect proper collision with the character and walls, and it wasn't there for an unknown reason. Copied the values from the Jam project.

NightAngel47 committed:
Added Collision Tests for Character Movement

Added tests to make sure that we know if collisoin with walls breaks for the CharacterMovement script.

NightAngel47 committed:
Added Werewolf Tests and Fixed issues with Setup and TearDown for All Tests

Added tsets for the WerewolfMovement script. Fixed issues with setup and teardown for tests that were messing up other tests.

clinealex10 reviewed (approved):
Looks good to me.
